### PR TITLE
Fix vulnerabilities. Bump material-ui and any other libs versions

### PR DESCRIPTION
**Pull request by Nikolo**
# Update deprecated and vulnerabilities dependencies

- Material-ui v4 -> v5
- react 16.14.2 -> 17.0.74

# Vulnerabilities stat
Before update:

```bash
% npm audit --omit=dev
16 vulnerabilities (6 moderate, 7 high, 3 critical)
```

After update:
```bash
% npm audit --omit=dev                                
found 0 vulnerabilities
```

### CODE REVIEW

**Comment by aleksey-mashanov** [comment]
Wow! I will look at this closely a little bit later.
The only thing I want to comment right now is that admin/static/lib is third party libraries and better not to touch them at all.

**Comment by Nikolo** [comment]
> Wow! I will look at this closely a little bit later. The only thing I want to comment right now is that admin/static/lib is third party libraries and better not to touch them at all.

It's autocorrection. I revert all changes in static/lib thanks!

**Comment by aleksey-mashanov** [comment]
Please, test this PR carefully. It simply doesn't work as expected: UI is broken.

P.S. MUI upgrading usually is not an easy task. It requires days of extensive small modifications and careful testing of every UI component.

**Review by aleksey-mashanov — CHANGES_REQUESTED** [review]

**Comment by aleksey-mashanov** [comment]
The problem of the broken UI can be as simple as forgotten step in the migration process. [This](https://mui.com/material-ui/migration/v5-style-changes/#%E2%9C%85-adjust-css-injection-order), for example.
But usually not. [Here](https://github.com/onlineconf/onlineconf/commit/5c36328d404ccb31ffcb085d5f05ddd497d67dc4) for reference is commit that upgraded MUI from v3 to v4.

File: admin/js/src/components/Editor.tsx [hunk 1]
```diff
@@ -208,4 +207,4 @@ class Editor extends React.Component<EditorProps & WithStyles<typeof styles> & W
 
 }
 
-export default withStyles(styles)(withMobileDialog<EditorProps>()(withTranslation()(Editor)));
+export default withStyles(styles)(withTranslation()(Editor));
```
[the remark this answers is not in the document]
Comment (Nikolo) [reply admin/js/src/components/Editor.tsx]: Fixed

File: admin/js/src/components/ParamAccess.tsx [hunk 2]
```diff
@@ -207,4 +207,4 @@ class ParamAccess extends React.Component<ParamDialogProps & WithStyles<typeof s
 
 }
 
-export default withStyles(styles)(withMobileDialog<ParamDialogProps>()(withTranslation()(ParamAccess)));
+export default withStyles(styles)(withTranslation()(ParamAccess));
```
[the remark this answers is not in the document]
Comment (Nikolo) [reply admin/js/src/components/ParamAccess.tsx]: Fixed

File: admin/js/src/components/Viewer.tsx [hunk 3]
```diff
@@ -60,4 +60,4 @@ const Viewer = ({ param, classes, t, ...props }: ViewerProps & WithStyles<typeof
 	</Dialog>
 );
 
-export default withStyles(styles)(withMobileDialog<ViewerProps>()(withTranslation()(Viewer)));
+export default withStyles(styles)(withTranslation()(Viewer));
```
[the remark this answers is not in the document]
Comment (Nikolo) [reply admin/js/src/components/Viewer.tsx]: Fixed

**Comment by aleksey-mashanov** [comment]
The UI looks good. I'll look at the code a little bit later.

**Review by aleksey-mashanov — COMMENTED** [review]
LGTM
Waiting for the approval from the QA.

**Review by aleksey-mashanov — APPROVED** [review]

**Pull request merged**